contracts/bancorx/BancorX.sol

```diff
@@ -34,395 +34,577 @@ contract BancorX is IBancorX, TokenHolder, ContractRegistryClient {
+            bytes32,
+            address,
+            uint8,
+            bool
+        )
+    {
+        Transaction memory transaction = _transactions[txId];
+
+        return (
+            transaction.amount,
+            transaction.fromBlockchain,
+            transaction.to,
+            transaction.numOfReports,
+            transaction.completed
+        );
+    }
+
+    /**
+     * @dev returns the transaction ID corresponding to provided xTransfer ID
+     *
+     * @param xTransferID xTransfer ID
+     *
+     * @return the transaction ID corresponding to provided xTransfer ID
+     */
+    function transactionIds(uint256 xTransferID) external view returns (uint256) {
+        return _transactionIds[xTransferID];
+    }
+
+    /**
+     * @dev returns whether a provided reported has already reported a txId
+     *
+     * @param txId the transaction ID to check
+     * @param reporter the report to check
+     *
+     * @return whether a provided reported has already reported a txId
+     */
+    function transactionIds(uint256 txId, address reporter) external view returns (bool) {
+        return _reportedTxs[txId][reporter];
     }
 
     /**
      * @dev upgrades the contract to the latest version
      * can only be called by the owner
      * note that the owner needs to call acceptOwnership on the new contract after the upgrade
      *
-     * @param _reporters    new list of reporters
+     * @param newReporters new list of reporters
      */
-    function upgrade(address[] memory _reporters) public ownerOnly {
+    function upgrade(address[] memory newReporters) external ownerOnly {
         IBancorXUpgrader bancorXUpgrader = IBancorXUpgrader(addressOf(BANCOR_X_UPGRADER));
 
         transferOwnership(address(bancorXUpgrader));
-        bancorXUpgrader.upgrade(version, _reporters);
+
+        bancorXUpgrader.upgrade(VERSION, newReporters);
+
         acceptOwnership();
     }
 
     /**
      * @dev claims tokens from msg.sender to be converted to tokens on another blockchain
      *
-     * @param _toBlockchain    blockchain on which tokens will be issued
-     * @param _to              address to send the tokens to
-     * @param _amount          the amount of tokens to transfer
+     * @param toBlockchain blockchain on which tokens will be issued
+     * @param to address to send the tokens to
+     * @param amount the amount of tokens to transfer
      */
     function xTransfer(
-        bytes32 _toBlockchain,
-        bytes32 _to,
-        uint256 _amount
-    ) public xTransfersAllowed {
+        bytes32 toBlockchain,
+        bytes32 to,
+        uint256 amount
+    ) external xTransfersAllowed {
         // get the current lock limit
         uint256 currentLockLimit = getCurrentLockLimit();
 
         // verify lock limit
-        require(_amount >= minLimit && _amount <= currentLockLimit, "ERR_AMOUNT_TOO_HIGH");
+        require(amount >= _minLimit && amount <= currentLockLimit, "ERR_AMOUNT_TOO_HIGH");
```

Comment:
suggestion: `_minLimit <= amount && amount <= currentLockLimit`

Reply:
BTW, the error-message doesn't seem entire accurate for this condition

Reply:
I'll switch the condition, but let's keep the error message as-is for now. It's less critical, since it's BancorX